src/Mesh/babylon.mesh.vertexData.ts

```diff
@@ -8,23 +11,50 @@
         setIndices(indices: IndicesArray, totalVertices: Nullable<number>, updatable?: boolean): void;
     }
 
+    /**
+     * This class contains the various kinds of data on every vertex of a mesh used in determining its shape and appearance 
+     */
     export class VertexData {
+        /**
+         * A linear array of the the x, y, z position of each vertex  [...., x, y, z, .....]
+         */
         public positions: Nullable<FloatArray>;
+        /**
+         * A linear array of the the x, y, z normal vector of each vertex  [...., x, y, z, .....]
+         */
         public normals: Nullable<FloatArray>;
+        /**
+         * A linear array of the the x, y, z tangent vector of each vertex  [...., x, y, z, .....]
+         */
         public tangents: Nullable<FloatArray>;
+        /**
+         * A linear array of the the u,v mapping of the texture image of each vertex  [...., u, v, .....]
+         */
         public uvs: Nullable<FloatArray>;
         public uvs2: Nullable<FloatArray>;
         public uvs3: Nullable<FloatArray>;
         public uvs4: Nullable<FloatArray>;
         public uvs5: Nullable<FloatArray>;
         public uvs6: Nullable<FloatArray>;
+        /**
+         * A linear array of the r, g, b, color of each vertex  [...., r, g, b, a, .....]
+         */
         public colors: Nullable<FloatArray>;
         public matricesIndices: Nullable<FloatArray>;
         public matricesWeights: Nullable<FloatArray>;
         public matricesIndicesExtra: Nullable<FloatArray>;
         public matricesWeightsExtra: Nullable<FloatArray>;
+        /**

```

Comment:
Matrices indices is the index in the array of matrices produced by bones. Each vertex have up to 4 indices (8 if the mstricesIndicesExtra is set).
Weights defines the weight of each indexed matrix in the final computation